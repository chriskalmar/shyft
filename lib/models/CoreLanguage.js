
import {
  dataTypes,
  Entity,
} from '../../';


const {
  DataTypeString,
} = dataTypes



export const CoreLanguage = new Entity({
  name: 'language',
  domain: 'core',
  description: 'A language',

  // indexing: {
  //   unique: [
  //     [ 'name' ],
  //     [ 'iso_code' ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      maxLength: 30,
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      maxLength: 30,
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      pattern: '^[a-z]+$',
      minLength: 2,
      maxLength: 2,
      required: true
    },
  }

})

