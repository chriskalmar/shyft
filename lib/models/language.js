
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'language',
      description: 'a language',

      cascadeOnDelete: true,

      attributes: [

        {
          name: 'name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'native_name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'iso_code',
          type: 'string',
          description: 'ISO 639-1 code of the language',
          maxLength: 2,
          required: true
        },

      ],

      indexing: {
        unique: [
          [ 'name' ],
          [ 'iso_code' ],
        ]
      }

    }

  ]

}
