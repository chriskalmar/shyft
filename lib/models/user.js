
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'user',
      description: 'a user',

      isUserTable: true,
      cascadeOnDelete: true,

      attributes: [

        {
          name: 'username',
          type: 'string',
          description: 'Username',
          pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
          minLength: 3,
          maxLength: 30,
          required: true
        },

        {
          name: 'first_name',
          type: 'string',
          description: 'First name',
          maxLength: 30
        },

        {
          name: 'last_name',
          type: 'string',
          description: 'Last name',
          maxLength: 30
        },

        {
          name: 'language',
          type: 'reference',
          description: 'User language',
          target: 'core::language',
          required: true
        },

        {
          name: 'email',
          type: 'email',
          description: 'Email address',
          required: true
        },

        {
          name: 'password',
          type: 'password',
          description: 'User password',
          required: true
        }
      ],

      indexing: {
        unique: [
          [ 'username' ],
          [ 'email' ],
        ]
      }

    }

  ]

}
