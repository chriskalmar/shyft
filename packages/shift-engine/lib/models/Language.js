
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';


export const Language = new Entity({
  name: 'Language',
  description: 'A language',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'name' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'isoCode' ],
    }),
  ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      required: true
    },
  }

})

