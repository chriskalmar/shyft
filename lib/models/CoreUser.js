
import {
  datatypes,
  Entity,
} from '../../';

import CoreLanguage from './CoreLanguage';

const {
  DataTypeString,
} = datatypes

export const CoreUser = new Entity({
  name: 'user',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  // indexing: {
  //   unique: [
  //     [ 'username' ],
  //     [ 'email' ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: CoreLanguage,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



