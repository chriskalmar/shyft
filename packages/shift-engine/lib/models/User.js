
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';


export const User = new Entity({
  name: 'User',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexing: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: Language,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



