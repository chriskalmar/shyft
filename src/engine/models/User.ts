import { Entity } from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import { Index, INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';

export const User = new Entity({
  name: 'User',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],

  attributes: {
    username: {
      type: DataTypeString,
      description: 'Username',
      required: true,
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
    },

    language: {
      type: Language,
      description: 'User language',
      required: true,
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true,
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true,
    },
  },
});
