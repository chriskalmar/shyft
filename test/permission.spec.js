import './setupAndTearDown';
import {
  find,
  count,
} from './db';

import {
  asUser,
  removeListDynamicData,
  asyncForEach,
} from './testUtils';


import { Board } from './models/Board';
import { Participant } from './models/Participant';



const orderByIdAsc = {
  orderBy: [{
    attribute: 'id',
    direction: 'ASC'
  }]
}


describe('permission', () => {
  describe('story', () => {

    it('a user should only see public boards and boards the user was invited to', async () => {

      const userIds = [46, 69, 40]

      await asyncForEach(userIds, async userId => {
        const result = await find(Board, { ...orderByIdAsc }, asUser(userId))
        result.data = removeListDynamicData(Board, result.data)
        expect(result).toMatchSnapshot('list')

        const rowCount = await count(Board, { }, asUser(userId))
        expect(rowCount).toMatchSnapshot('count')
      })

      const filter = {
        isPrivate: true
      }

      await asyncForEach(userIds, async userId => {
        const result = await find(Board, { ...orderByIdAsc, filter }, asUser(userId))
        result.data = removeListDynamicData(Board, result.data)
        expect(result).toMatchSnapshot('private list')

        const rowCount = await count(Board, { filter }, asUser(userId))
        expect(rowCount).toMatchSnapshot('private count')
      })
    })


    it('a user should only see participants of boards the user is part of', async () => {

      const userIds = [46, 69, 40]

      await asyncForEach(userIds, async userId => {
        const result = await find(Participant, { ...orderByIdAsc }, asUser(userId))
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot('list')

        const rowCount = await count(Participant, {}, asUser(userId))
        expect(rowCount).toMatchSnapshot('count')
      })

      await asyncForEach(userIds, async userId => {

        const filter = {
          invitee: {
            $ne: userId
          }
        }

        const result = await find(Participant, { ...orderByIdAsc, filter }, asUser(userId))
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot('list')

        const rowCount = await count(Participant, { filter }, asUser(userId))
        expect(rowCount).toMatchSnapshot('count')
      })

    })


  })

})
