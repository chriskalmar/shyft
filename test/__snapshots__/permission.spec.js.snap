// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`permission story a user should only see participants of boards the user is part of: count 1`] = `23`;

exports[`permission story a user should only see participants of boards the user is part of: count 2`] = `15`;

exports[`permission story a user should only see participants of boards the user is part of: count 3`] = `43`;

exports[`permission story a user should only see participants of boards the user is part of: count 4`] = `19`;

exports[`permission story a user should only see participants of boards the user is part of: count 5`] = `13`;

exports[`permission story a user should only see participants of boards the user is part of: count 6`] = `36`;

exports[`permission story a user should only see participants of boards the user is part of: list 1`] = `
Object {
  "data": Array [
    Object {
      "board": "16",
      "id": "9",
      "invitee": "93",
      "inviter": "93",
      "state": 50,
    },
    Object {
      "board": "16",
      "id": "14",
      "invitee": "92",
      "inviter": "92",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "76",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
    Object {
      "board": "16",
      "id": "107",
      "invitee": "46",
      "inviter": "46",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "150",
      "invitee": "69",
      "inviter": "69",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "152",
      "invitee": "46",
      "inviter": "46",
      "state": 50,
    },
    Object {
      "board": "16",
      "id": "177",
      "invitee": "5",
      "inviter": "5",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "196",
      "invitee": "25",
      "inviter": "25",
      "state": 50,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see participants of boards the user is part of: list 2`] = `
Object {
  "data": Array [
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "43",
      "invitee": "80",
      "inviter": "80",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "57",
      "invitee": "28",
      "inviter": "28",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "75",
      "invitee": "21",
      "inviter": "21",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "76",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "77",
      "invitee": "81",
      "inviter": "81",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "101",
      "invitee": "69",
      "inviter": "69",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "113",
      "invitee": "55",
      "inviter": "55",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "144",
      "invitee": "99",
      "inviter": "99",
      "state": 50,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see participants of boards the user is part of: list 3`] = `
Object {
  "data": Array [
    Object {
      "board": "13",
      "id": "5",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "36",
      "id": "7",
      "invitee": "81",
      "inviter": "81",
      "state": 50,
    },
    Object {
      "board": "13",
      "id": "30",
      "invitee": "79",
      "inviter": "79",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "36",
      "id": "32",
      "invitee": "41",
      "inviter": "41",
      "state": 50,
    },
    Object {
      "board": "18",
      "id": "35",
      "invitee": "12",
      "inviter": "12",
      "state": 50,
    },
    Object {
      "board": "13",
      "id": "37",
      "invitee": "64",
      "inviter": "64",
      "state": 50,
    },
    Object {
      "board": "18",
      "id": "48",
      "invitee": "44",
      "inviter": "44",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "76",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see participants of boards the user is part of: list 4`] = `
Object {
  "data": Array [
    Object {
      "board": "16",
      "id": "9",
      "invitee": "93",
      "inviter": "93",
      "state": 50,
    },
    Object {
      "board": "16",
      "id": "14",
      "invitee": "92",
      "inviter": "92",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "76",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "150",
      "invitee": "69",
      "inviter": "69",
      "state": 50,
    },
    Object {
      "board": "16",
      "id": "177",
      "invitee": "5",
      "inviter": "5",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "196",
      "invitee": "25",
      "inviter": "25",
      "state": 50,
    },
    Object {
      "board": "9",
      "id": "202",
      "invitee": "7",
      "inviter": "52",
      "state": 20,
    },
    Object {
      "board": "9",
      "id": "208",
      "invitee": "58",
      "inviter": "52",
      "state": 20,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see participants of boards the user is part of: list 5`] = `
Object {
  "data": Array [
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "43",
      "invitee": "80",
      "inviter": "80",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "57",
      "invitee": "28",
      "inviter": "28",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "75",
      "invitee": "21",
      "inviter": "21",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "76",
      "invitee": "40",
      "inviter": "40",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "77",
      "invitee": "81",
      "inviter": "81",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "113",
      "invitee": "55",
      "inviter": "55",
      "state": 50,
    },
    Object {
      "board": "31",
      "id": "144",
      "invitee": "99",
      "inviter": "99",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "152",
      "invitee": "46",
      "inviter": "46",
      "state": 50,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see participants of boards the user is part of: list 6`] = `
Object {
  "data": Array [
    Object {
      "board": "36",
      "id": "7",
      "invitee": "81",
      "inviter": "81",
      "state": 50,
    },
    Object {
      "board": "13",
      "id": "30",
      "invitee": "79",
      "inviter": "79",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "31",
      "invitee": "57",
      "inviter": "57",
      "state": 50,
    },
    Object {
      "board": "36",
      "id": "32",
      "invitee": "41",
      "inviter": "41",
      "state": 50,
    },
    Object {
      "board": "18",
      "id": "35",
      "invitee": "12",
      "inviter": "12",
      "state": 50,
    },
    Object {
      "board": "13",
      "id": "37",
      "invitee": "64",
      "inviter": "64",
      "state": 50,
    },
    Object {
      "board": "18",
      "id": "48",
      "invitee": "44",
      "inviter": "44",
      "state": 50,
    },
    Object {
      "board": "14",
      "id": "85",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
    Object {
      "board": "36",
      "id": "117",
      "invitee": "9",
      "inviter": "9",
      "state": 50,
    },
    Object {
      "board": "18",
      "id": "120",
      "invitee": "95",
      "inviter": "95",
      "state": 50,
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: count 1`] = `33`;

exports[`permission story a user should only see public boards and boards the user was invited to: count 2`] = `30`;

exports[`permission story a user should only see public boards and boards the user was invited to: count 3`] = `34`;

exports[`permission story a user should only see public boards and boards the user was invited to: list 1`] = `
Object {
  "data": Array [
    Object {
      "createdBy": "37",
      "id": "1",
      "isPrivate": false,
      "name": "Reiciendis quaerat",
      "owner": "37",
      "updatedBy": "37",
    },
    Object {
      "createdBy": "46",
      "id": "2",
      "isPrivate": true,
      "name": "Et eum",
      "owner": "46",
      "updatedBy": "46",
    },
    Object {
      "createdBy": "42",
      "id": "4",
      "isPrivate": false,
      "name": "Delectus qui maxime",
      "owner": "42",
      "updatedBy": "42",
    },
    Object {
      "createdBy": "52",
      "id": "9",
      "isPrivate": true,
      "name": "Veritatis nihil cum",
      "owner": "52",
      "updatedBy": "52",
    },
    Object {
      "createdBy": "81",
      "id": "11",
      "isPrivate": false,
      "name": "Aut incidunt consequatur",
      "owner": "81",
      "updatedBy": "81",
    },
    Object {
      "createdBy": "23",
      "id": "12",
      "isPrivate": false,
      "name": "Tenetur porro",
      "owner": "23",
      "updatedBy": "23",
    },
    Object {
      "createdBy": "84",
      "id": "13",
      "isPrivate": false,
      "name": "Suscipit aut cum",
      "owner": "84",
      "updatedBy": "84",
    },
    Object {
      "createdBy": "26",
      "id": "14",
      "isPrivate": false,
      "name": "Id cum",
      "owner": "26",
      "updatedBy": "26",
    },
    Object {
      "createdBy": "60",
      "id": "15",
      "isPrivate": false,
      "name": "Sunt nobis",
      "owner": "60",
      "updatedBy": "60",
    },
    Object {
      "createdBy": "73",
      "id": "16",
      "isPrivate": false,
      "name": "Est adipisci inventore",
      "owner": "73",
      "updatedBy": "73",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: list 2`] = `
Object {
  "data": Array [
    Object {
      "createdBy": "37",
      "id": "1",
      "isPrivate": false,
      "name": "Reiciendis quaerat",
      "owner": "37",
      "updatedBy": "37",
    },
    Object {
      "createdBy": "42",
      "id": "4",
      "isPrivate": false,
      "name": "Delectus qui maxime",
      "owner": "42",
      "updatedBy": "42",
    },
    Object {
      "createdBy": "81",
      "id": "11",
      "isPrivate": false,
      "name": "Aut incidunt consequatur",
      "owner": "81",
      "updatedBy": "81",
    },
    Object {
      "createdBy": "23",
      "id": "12",
      "isPrivate": false,
      "name": "Tenetur porro",
      "owner": "23",
      "updatedBy": "23",
    },
    Object {
      "createdBy": "84",
      "id": "13",
      "isPrivate": false,
      "name": "Suscipit aut cum",
      "owner": "84",
      "updatedBy": "84",
    },
    Object {
      "createdBy": "26",
      "id": "14",
      "isPrivate": false,
      "name": "Id cum",
      "owner": "26",
      "updatedBy": "26",
    },
    Object {
      "createdBy": "60",
      "id": "15",
      "isPrivate": false,
      "name": "Sunt nobis",
      "owner": "60",
      "updatedBy": "60",
    },
    Object {
      "createdBy": "73",
      "id": "16",
      "isPrivate": false,
      "name": "Est adipisci inventore",
      "owner": "73",
      "updatedBy": "73",
    },
    Object {
      "createdBy": "24",
      "id": "17",
      "isPrivate": false,
      "name": "Est et",
      "owner": "24",
      "updatedBy": "24",
    },
    Object {
      "createdBy": "75",
      "id": "18",
      "isPrivate": false,
      "name": "In ipsum est",
      "owner": "75",
      "updatedBy": "75",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: list 3`] = `
Object {
  "data": Array [
    Object {
      "createdBy": "37",
      "id": "1",
      "isPrivate": false,
      "name": "Reiciendis quaerat",
      "owner": "37",
      "updatedBy": "37",
    },
    Object {
      "createdBy": "46",
      "id": "2",
      "isPrivate": true,
      "name": "Et eum",
      "owner": "46",
      "updatedBy": "46",
    },
    Object {
      "createdBy": "42",
      "id": "4",
      "isPrivate": false,
      "name": "Delectus qui maxime",
      "owner": "42",
      "updatedBy": "42",
    },
    Object {
      "createdBy": "81",
      "id": "11",
      "isPrivate": false,
      "name": "Aut incidunt consequatur",
      "owner": "81",
      "updatedBy": "81",
    },
    Object {
      "createdBy": "23",
      "id": "12",
      "isPrivate": false,
      "name": "Tenetur porro",
      "owner": "23",
      "updatedBy": "23",
    },
    Object {
      "createdBy": "84",
      "id": "13",
      "isPrivate": false,
      "name": "Suscipit aut cum",
      "owner": "84",
      "updatedBy": "84",
    },
    Object {
      "createdBy": "26",
      "id": "14",
      "isPrivate": false,
      "name": "Id cum",
      "owner": "26",
      "updatedBy": "26",
    },
    Object {
      "createdBy": "60",
      "id": "15",
      "isPrivate": false,
      "name": "Sunt nobis",
      "owner": "60",
      "updatedBy": "60",
    },
    Object {
      "createdBy": "73",
      "id": "16",
      "isPrivate": false,
      "name": "Est adipisci inventore",
      "owner": "73",
      "updatedBy": "73",
    },
    Object {
      "createdBy": "24",
      "id": "17",
      "isPrivate": false,
      "name": "Est et",
      "owner": "24",
      "updatedBy": "24",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: private count 1`] = `3`;

exports[`permission story a user should only see public boards and boards the user was invited to: private count 2`] = `0`;

exports[`permission story a user should only see public boards and boards the user was invited to: private count 3`] = `4`;

exports[`permission story a user should only see public boards and boards the user was invited to: private list 1`] = `
Object {
  "data": Array [
    Object {
      "createdBy": "46",
      "id": "2",
      "isPrivate": true,
      "name": "Et eum",
      "owner": "46",
      "updatedBy": "46",
    },
    Object {
      "createdBy": "52",
      "id": "9",
      "isPrivate": true,
      "name": "Veritatis nihil cum",
      "owner": "52",
      "updatedBy": "52",
    },
    Object {
      "createdBy": "26",
      "id": "49",
      "isPrivate": true,
      "name": "Odit qui",
      "owner": "26",
      "updatedBy": "26",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": false,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: private list 2`] = `
Object {
  "data": Array [],
  "pageInfo": Object {
    "hasNextPage": false,
    "hasPreviousPage": false,
  },
}
`;

exports[`permission story a user should only see public boards and boards the user was invited to: private list 3`] = `
Object {
  "data": Array [
    Object {
      "createdBy": "46",
      "id": "2",
      "isPrivate": true,
      "name": "Et eum",
      "owner": "46",
      "updatedBy": "46",
    },
    Object {
      "createdBy": "30",
      "id": "24",
      "isPrivate": true,
      "name": "Non iusto unde",
      "owner": "30",
      "updatedBy": "30",
    },
    Object {
      "createdBy": "47",
      "id": "33",
      "isPrivate": true,
      "name": "Suscipit quas",
      "owner": "47",
      "updatedBy": "47",
    },
    Object {
      "createdBy": "63",
      "id": "46",
      "isPrivate": true,
      "name": "Et architecto",
      "owner": "63",
      "updatedBy": "63",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": false,
    "hasPreviousPage": false,
  },
}
`;
